lib/flowViewport/sceneIndex/fvpWireframeSelectionHighlightSceneIndex.cpp

```diff
@@ -791,125 +963,173 @@ WireframeSelectionHighlightSceneIndex::_CollectSelectionHighlightMirrors(const P
     }
     outSelectionHighlightMirrors.insert(selectionHighlightPrimPath);
 
-    // Traverse the children of this prim to find the child instancers, and add its instancing-related
-    // paths so we can process them and create selection highlight mirrors for them as well.
-    SdfPathVector affectedOriginalPrimPaths;
+    // Traverse the children of this prim to find the affected child prims, and process their instancing-related
+    // paths so we can create selection highlight mirrors for these prims as well.
+    SdfPathVector affectedPrototypePaths;
+    SdfPathVector affectedInstancedByPaths;
     auto operation = [&](const SdfPath& primPath, const HdSceneIndexPrim& prim) -> bool {
         outAddedPrims.push_back({primPath.ReplacePrefix(originalPrimPath, selectionHighlightPrimPath), prim.primType});
-        if (prim.primType == HdPrimTypeTokens->instancer) {
-            SdfPathVector instancingRelatedPaths = _GetInstancingRelatedPaths(prim);
-            affectedOriginalPrimPaths.insert(affectedOriginalPrimPaths.end(), instancingRelatedPaths.begin(), instancingRelatedPaths.end());
-            return false; // We have found an instancer, don't process its children (nested instancers will be processed through the instancing-related paths).
+        if (prim.primType == HdPrimTypeTokens->instancer || prim.primType == HdPrimTypeTokens->mesh) {
+            if (direction & SelectionHighlightsCollectionDirection::Prototypes) {
```

Comment:
Splitting the instancing-related paths in two prevents us from always traversing the entire instancing graph, and inadvertently creating selection highlights for non-selected prims. Without this, selecting an instanced mesh would create a highlight mirror for the mesh, then a highlight mirror for the instancer, but in creating the highlight mirror for the instancer and collecting its instancing paths, we would collect its other prototype paths, and then create highlight mirrors for those as well, even if they weren't actually selected.